xtask/src/main.rs

```diff
@@ -74,24 +74,39 @@ fn main() -> Result<()> {
     }
     let chosen_src = fs::read_to_string(&chosen_file)?;
     let chosen_doc: DocumentMut = chosen_src.parse()?;
-    if let Some(tbl) = chosen_doc
+    let Some(patches) = chosen_doc
         .get("patch")
         .and_then(|p| p.get("crates-io"))
         .and_then(Item::as_table)
-    {
-        let ci = root_doc["patch"]["crates-io"].as_table_mut().unwrap();
-        for (k, v) in tbl {
-            ci[k] = v.clone(); // overwrite / insert
-        }
-    } else {
+    else {
         bail!("{} has no [patch.crates-io] table", chosen_file.display());
+    };
+    for (k, v) in patches {
+        root_doc["patch"]["crates-io"].as_table_mut().unwrap()[k] = v.clone(); // overwrite / insert
     }
 
     // 4 ── write the updated manifest back
     fs::write(&manifest_path, root_doc.to_string())
         .with_context(|| format!("writing {}", manifest_path.display()))?;
 
-    // 5 ── forward to Cargo
+    // 5 ── run cargo update for patching crates
+    let status = Command::new("cargo")
```

Comment:
This was mainly to update Cargo.lock when applying patches instead of when it is first run/compiled?

Reply:
Yes, not sure why the patches of Airbender will be marked as unused in `Cargo.lock` if we don't run `cargo update` explicitly, but other patches seems working fine